src/cdk-experimental/menu/focus-mouse-manager.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {QueryList, ElementRef} from '@angular/core';
+import {fromEvent, Observable} from 'rxjs';
+import {mapTo, mergeAll, takeUntil, startWith, mergeMap} from 'rxjs/operators';
+
+/** Item to track for mouse focus events. */
+export interface FocusableElement {
+  /** A reference to the element to be tracked. */
+  _elementRef: ElementRef<HTMLElement>;
+}
+
+/**
+ * FocusMouseManager keeps track of the element under mouse focus. It emits the element on an
+ * observable when an element is moused into.
+ */
+export class FocusMouseManager<T extends FocusableElement> {
+  /** Observable which emits when the mouse enters one of the managed elements. */
+  readonly mouseEntered: Observable<T> = this._getMouseEnterObservable();
+
+  constructor(private readonly _elements: QueryList<T>) {}
+
+  /**
```

Comment:
Currently, a new observable is generated every time the method is called. Is this strictly necessary, or can the same purpose be served by using a single observable instance?

If a new observable is required every time, you should explain why in the documentation. (I would also rename it to '_getNewMouseEnterObservable()' to make this clear.)

Reply:
Good point. It was part of the construction of the observable so it was only called once. However with Jeremys comment below it'll just end up as a function so it should be addressed